modules/components/overridable-props/overridable-prop-parser.php

```diff
@@ -68,17 +66,76 @@ private function validate( array $prop ): Parse_Result {
 	private function sanitize( array $prop ): Parse_Result {
 		$result = Parse_Result::make();
 
+		$sanitized_origin_value = $this->get_sanitized_origin_value( $prop );
+
 		$sanitized_prop = [
 			'overrideKey' => sanitize_key( $prop['overrideKey'] ),
 			'label' => sanitize_text_field( $prop['label'] ),
 			'elementId' => sanitize_key( $prop['elementId'] ),
 			'propKey' => sanitize_text_field( $prop['propKey'] ),
 			'widgetType' => sanitize_text_field( $prop['widgetType'] ),
 			'elType' => sanitize_text_field( $prop['elType'] ),
-			'originValue' => $this->origin_value_prop_type->sanitize( $prop['originValue'] ),
+			'originValue' => $sanitized_origin_value,
 			'groupId' => sanitize_key( $prop['groupId'] ),
 		];
 
+		if ( $this->is_with_origin_prop_fields( $prop ) ) {
+			$sanitized_prop['originPropFields'] = $prop['originPropFields'];
+		}
+
 		return $result->wrap( $sanitized_prop );
 	}
+
+	private function is_with_origin_prop_fields( array $prop ): bool {
+		return ! empty( $prop['originPropFields'] );
+	}
+
+	private function get_origin_prop_type( array $prop ) {
+		if ( $this->is_with_origin_prop_fields( $prop ) ) {
+			return $this->get_origin_prop_type( $prop['originPropFields'] );
+		}
+
+		return Parsing_Utils::get_prop_type(
+			$prop['elType'],
+			$prop['widgetType'],
+			$prop['propKey'],
+		);
+	}
+
+	private function get_origin_value( array $prop ) {
+		if ( empty( $prop ) || empty( $prop['originValue'] ) ) {
+			return null;
+		}
+
+		if ( isset( $prop['originValue']['$$type'] ) && Override_Prop_Type::get_key() === $prop['originValue']['$$type'] ) {
+			return $prop['originValue']['value']['override_value'];
+		}
+
+		return $prop['originValue'];
+	}
+
+	private function get_sanitized_origin_value( array $prop ) {
+		$origin_value = $this->get_origin_value( $prop );
+		$origin_prop_type = $this->get_origin_prop_type( $prop );
+
+		if ( ! empty( $origin_value ) ) {
+			$sanitized_value = $origin_prop_type->sanitize( $origin_value );
+
+			if ( Override_Prop_Type::get_key() === $origin_value['$$type'] ) {
```

Comment:
Shouldn't we check `$prop['originValue']['$$type']`?
```suggestion
			if ( Override_Prop_Type::get_key() === $prop['originValue']['$$type'] ) {
```